### PR TITLE
use f-strings instead of format

### LINKED ISSUE #692: use f-strings wherever possible

### PR DESCRIPTION
**Pull request by codeboten**
# Description

Continuing work to support pylint 2.11.0, addressing string formatting in code.

Fixes #692

### CODE REVIEW

**Review by lzchen — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py [hunk 1]
```diff
@@ -349,10 +349,7 @@ def test_trace_response_headers(self):
         )
         self.assertEqual(
             response.headers["traceresponse"],
-            "00-{0}-{1}-01".format(
```
Comment (aabmass) [inline instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py:352]: Cases like these where you don't have short variable, f-strings can be pretty ugly :(

Comment (codeboten) [reply instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py:352]: I hear that... I don't feel strongly either ways. I found the formatting previously there confusing as well w/ the multiple 0's and 1's 😄 

Could create a temporary variable or just disable pylint for this line.

**Review by ocelotl — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py (hunk 1 above)
Comment (alertedsnake) [reply instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py:352]: It's probably more readable when it's not complex code buried in a string - I prefer the format one.

**Review by owais — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py (hunk 1 above)
Comment (aabmass) [reply instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py:352]: > I found the formatting previously there confusing as well w/ the multiple 0's and 1's

I agree, maybe short variable names with f-strings would be better here, or could do `"00-{}-{}-01".format(...)` or named format params. I'll leave it up to you, don't want to block this @codeboten

Comment (codeboten) [reply instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py:352]: i'm happy to fix this in a following PR. I will refactor all the tests that are duplicated into a single test in the base class.

Comment (codeboten) [reply instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py:352]: will use the following method to simplify the code https://github.com/open-telemetry/opentelemetry-python/pull/2159

**Pull request merged**